### PR TITLE
Add translator for sverigesradio.se

### PR DESCRIPTION
**Pull request by sebastian-berlin-wmse**
This is the website for the Swedish public service radio.

### CODE REVIEW

File: Sveriges radio.js [hunk 1]
```diff
@@ -0,0 +1,709 @@
+{
+	"translatorID": "caa8f42c-9dbf-446e-963b-6ee18e3133d2",
+	"label": "Sveriges radio",
+	"creator": "Sebastian Berlin",
+	"target": "^https?://sverigesradio\\.se/sida/artikel.aspx",
```
Comment (adam3smith) [inline Sveriges radio.js]: This doesn't seem right -- it wouldn't detect multiples if your code below is correct (please add some multiples as tests, if possible)

File: Sveriges radio.js [hunk 2]
```diff
@@ -0,0 +1,709 @@
...
+	// Embedded Metadata
+	translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
+	// translator.setDocument(doc);
+	
+	translator.setHandler('itemDone', function (obj, item) {
+		item.publicationTitle = "Sveriges Radio";
+
+		item.creators = [];
+		var nameNodes = ZU.xpath(doc, '//p[@class="byline"]/text()');
+		for(var node of nameNodes) {
```
Comment (adam3smith) [inline Sveriges radio.js]: `let` is prefered over `var` for loops and similar temp uses

File: Sveriges radio.js [hunk 3]
```diff
@@ -0,0 +1,709 @@
...
+			var nameString = node.wholeText;
+			if(nameString.trim() === "") {
+				// Empty text node are added between authors; we want to skip
+				// those.
+				continue;
+			}
+			var names = nameString.split(" ");
```
Comment (adam3smith) [inline Sveriges radio.js]: It seems like you're just generating a crude version of `ZU.cleanAuthor` here -- I'd recommend using that.

File: Sveriges radio.js [hunk 4]
```diff
@@ -0,0 +1,709 @@
...
+				lastName: lastName,
+				creatorType: "author"
+			};
+			item.creators.push(author);
+		}
+
+		// The title from the meta is in the format:
+		// Australierna säger ja till samkönade äktenskap - Nyheter (Ekot)
+		var titleString =
+			ZU.xpath(doc, '//meta[@name="twitter:title"]')[0]
```
Comment (adam3smith) [inline Sveriges radio.js]: use `attr` and CSS selector here instead for simplicity

File: Sveriges radio.js [hunk 5]
```diff
@@ -0,0 +1,709 @@
...
+			.getAttribute("content");
+		var titleParts = titleString.split(" - ");
+		item.title = titleParts[0];
+		item.section = titleParts[1];
+
+		var dateString = 
+			ZU.xpath(doc, '//meta[@name="displaydate"]')[0]
```
Comment (adam3smith) [inline Sveriges radio.js]: same here

File: Sveriges radio.js [hunk 6]
```diff
@@ -0,0 +1,709 @@
...
+			.getAttribute("content");
+		item.date = dateString.replace(/(....)(..)(..)/, "$1-$2-$3");
```
Comment (adam3smith) [inline Sveriges radio.js]: `(\d{4})(\d{2})(\d{2})` is more precise and more readable

File: Sveriges radio.js [hunk 7]
```diff
@@ -0,0 +1,709 @@
...
+
+		item.complete();
+	});
+
+	translator.getTranslatorObject(function(trans) {
+		trans.itemType = "newspaperArticle";
+		// TODO map additional meta tags here, or delete completely
```
Comment (adam3smith) [inline Sveriges radio.js]: remove all TODOs

File: Sveriges radio.js [hunk 8]
```diff
@@ -0,0 +1,709 @@
...
+	{
+		"type": "web",
+		"url": "https://sverigesradio.se/sida/artikel.aspx?programid=83&artikel=6821850",
+		"items": [
+			{
+				"itemType": "newspaperArticle",
+				"title": "Australierna säger ja till samkönade äktenskap",
+				"creators": [
+					{
+						"firstName": "Ekot",
```
Comment (adam3smith) [inline Sveriges radio.js]: using Zotero's built in function will solve this, but the right way to handle single-name and institutional authors is to use `lastName` only and set `fieldMode: true`

File: Sveriges radio.js [hunk 9]
```diff
@@ -0,0 +1,709 @@
...
+				"libraryCatalog": "sverigesradio.se",
+				"publicationTitle": "Sveriges Radio",
+				"section": "Nyheter (Ekot)",
+				"url": "http://sverigesradio.se/sida/artikel.aspx?programid=83&artikel=6821850",
+				"attachments": [
+					{
+						"title": "Snapshot"
+					}
+				],
+				"tags": [
```
Comment (adam3smith) [inline Sveriges radio.js]: These tags look generic and not helpful. I'd clean them out.

File: Sveriges radio.js (hunk 1 above)
Comment (sebastian-berlin-wmse) [reply Sveriges radio.js]: Extended regex to include search page.

File: Sveriges radio.js (hunk 3 above)
Comment (sebastian-berlin-wmse) [reply Sveriges radio.js]: I missed that function. Replaced the logic I had written, but some preprocessing was still required due to how the byline string can be formatted.

File: Sveriges radio.js (hunk 9 above)
Comment (sebastian-berlin-wmse) [reply Sveriges radio.js]: I wasn't sure what tags did, so I kept them as they were. As you say, at least some of them are quite general and don't seem to relate to the article. Happy to get rid of them.

**Comment by adam3smith** [comment]
Great, thanks!

**Pull request merged**